packages/flutter/lib/src/widgets/app.dart

```diff
@@ -301,19 +313,25 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
         ]
       );
     }
-    if (widget.showSemanticsDebugger) {
-      result = new SemanticsDebugger(
-        child: result
-      );
-    }
     assert(() {
+      if (widget.showSemanticsDebugger) {
+        result = new SemanticsDebugger(
+          child: result,
+        );
+      }
+      if (widget.showWidgetInspector || WidgetsApp.showWidgetInspectorOverride) {
+        result = new WidgetInspector (
+          child: result,
+        );
+      }
```

Comment:
if they're in the assert, that means they're only available in debug mode, in which case their names should start with `debug` and their docs should make this clear.

showSemanticsDebugger in particular is intended to work in profile builds too, so that one should stay outside the assert.